README.md

```diff
@@ -123,6 +123,45 @@ class User {
 export default createHandler(User);
 ```
 
+### Route matching
+
+It is possible to use Express.js style route matching within your handlers. To enable the functionality add the `path-to-regexp` package to your project by running:
+```bash
+$ yarn add path-to-regexp
+```
+
+Then you can define your routes in your handler like:
+```ts
+// pages/api/user/[[...api]].ts
+class User {
+  @Get()
+  public list() {
+    return DB.findAllUsers();
+  }
+
+  @Get('/:id')
+  public details(@Param('id') id: string) {
+    return DB.findUserById(id);
+  }
+
+  @Get('/:userId/comments')
+  public comments(@Param('userId') userId: string) {
+    return DB.findUserComments(userId);
+  }
+
+  @Get('/:userId/comments/:commentId')
+  public commentDetails(@Param('userId') userId: string, @Param('commentId') commentId: string) {
+    return DB.findUserCommentById(userId, commentId);
+  }
+}
+```
+
+💡 It is possible to use pipes with `@Param`. e.g: `@Param('userId', ParseNumberPipe) userId: number`
+
+⚠️ When `path-to-regexp` package is not installed and route matching is being used in handlers, the request will be handled by the method defined with the `/` path (keep in mind that using `@Get()` and `@Get('/')` do exactly the same thing).
+
+For the above example, a request to `api/user/123` will be handled by the `list` method if `path-to-regexp` package is not installed in your project.
+
```

Comment:
Might be nice to also add [this](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes), since it also comes with some gotcha's/caveats.